src/utils/gitHubUtils.ts

```diff
@@ -4,42 +4,35 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { Octokit } from '@octokit/rest';
-import { GitGetTreeResponseData, OctokitResponse, ReposGetBranchResponseData, UsersGetAuthenticatedResponseData } from '@octokit/types';
+import { GitGetTreeResponseData, OctokitResponse, ReposGetBranchResponseData, ReposGetResponseData, UsersGetAuthenticatedResponseData } from '@octokit/types';
 // tslint:disable-next-line:no-require-imports
 import gitUrlParse = require('git-url-parse');
-import { HttpMethods, IncomingMessage, TokenCredentials } from 'ms-rest';
-import { Response } from 'request';
 import * as git from 'simple-git/promise';
+import { URL } from 'url';
 import { authentication, QuickPickItem } from 'vscode';
 import { IAzureQuickPickItem } from 'vscode-azureextensionui';
 import { IStaticWebAppWizardContext } from '../commands/createStaticWebApp/IStaticWebAppWizardContext';
 import { createOctokitClient } from '../commands/github/createOctokitClient';
-import { githubApiEndpoint } from '../constants';
 import { GitTreeData, OrgForAuthenticatedUserData } from '../gitHubTypings';
 import { localize } from './localize';
 import { nonNullProp } from './nonNull';
-import { requestUtils } from './requestUtils';
 import { getSingleRootFsPath } from './workspaceUtils';
 
-// tslint:disable-next-line:no-reserved-keywords
-export type gitHubRepoData = { name: string; url: string; html_url: string; clone_url?: string; default_branch?: string };
-export type gitHubBranchData = { name: string };
-export type gitHubLink = { prev?: string; next?: string; last?: string; first?: string };
-export type gitHubWebResource = requestUtils.Request & { nextLink?: string };
+type gitHubLink = { prev?: string; next?: string; last?: string; first?: string };
 
-export async function getGitHubJsonResponse<T>(requestOptions: gitHubWebResource): Promise<T> {
+// tslint:disable:no-any no-unsafe-any
+export async function getGitHubJsonResponse<T, U>(gitHubApiCb: (params: U) => Promise<OctokitResponse<any>>, params: U & { page?: number }): Promise<T> {
     // Reference for GitHub REST routes
     // https://developer.github.com/v3/
-    // Note: blank after user implies look up authorized user
-    const gitHubResponse: Response = await requestUtils.sendRequest(requestOptions);
-    if (gitHubResponse.headers.link) {
-        const headerLink: string = <string>gitHubResponse.headers.link;
-        const linkObject: gitHubLink = parseLinkHeaderToGitHubLinkObject(headerLink);
-        requestOptions.nextLink = linkObject.next;
+    const res: OctokitResponse<any> = await gitHubApiCb(params);
+    if (res.headers.link) {
+        const linkObject: gitHubLink = parseLinkHeaderToGitHubLinkObject(res.headers.link);
+        const page: string | null | undefined = linkObject.next ? new URL(linkObject.next).searchParams.get('page') : undefined;
```

Comment:
Is 'page' officially supported/documented? I feel like I trust a next link more than I trust a page query param within a next link

Reply:
https://developer.github.com/v3/#link-header

The page is a query parameter for the API call itself.  Unless we keep using custom requests, I don't think I can use the next link that is retrieved directly anymore.

It's also annoyingly a number parameter (which is why I have to convert the query param from a string to a number).  I didn't see any other way to retrieve the next-page from the API response.